docs/developer_guide.md

```diff
@@ -51,7 +51,32 @@ select **MUXControlsTestApp** as your startup project in Visual Studio.
 
 ### Creating a NuGet package
 
-> More information will be coming on this soon
+To create a NuGet package for a given build flavor (release/debug) and build arch (x64/x86/ARM/ARM64), first you need to build the solution in that configuration.
+
+After building the solution in the desired configuration, you can run the `build-nupkg.ps1` script that will run the required steps.
+The script takes the following arguments:
+- BuildOutput: The path to the microsoft-ui-xaml BuildOutput folder
+- BuildFlavor (default: `release`): The flavor to use for nuget build (`debug` or `release`)
+- BuildArch (default `x86`): The build arch to use, one of: `x64`, `x86`, `ARM`, `ARM64`
+- OutputDir: The folder where the nuget package will be generated in
+- SkipFrameworkPackage (default `False`): Can be specified to skip building a framework package
+- SkipMakeNugetPackageAppxPackages (default `False`): Can be specified to skip generating appx files for the framework package
+
+Example usage (running from root of repository folder):
+```
+// Builds a NuGet package with debug binaries and arch x64 into the folder "NuGetPackage"
+.\build\NuSpecs\build-nupkg.ps1 -BuildOutput "..\..\BuildOutput" -BuildFlavor "debug" -BuildArch "x64" -OutputDir "..\..\NugetPackage"
+```
+
+> Note: To use debug packages, you need to change [this line](https://github.com/microsoft/microsoft-ui-xaml/blob/7d2cd793a0154580f1dd0c9685c461198e05f207/dev/dll/Microsoft.UI.Xaml.vcxproj#L35) in `microsoft-ui-xaml/dev/dll/Microsoft.UI.Xaml.vcxproj` from
```

Comment:
Could you clarify what you mean here?

Reply:
Added an explanation why that line needs to be changed,

Reply:
Interesting. I didn't know about that limitation.